### PR TITLE
feat: Functionality for sending event ready-to-settle emails

### LINKED ISSUE #373: Email notification for event settlement
Option to send an email notifying participants that the event is ready for settlement.

### PR DESCRIPTION
**Pull request by motionduck**
Implemented functionality to send emails notifying paying participants that an event is ready to be settled. An option for sending the email has been added to the event settings page.

Other changes:
- Fixed issue where join event button was visible for non-active events on mobile.
- Added password validation of minimum 3 characters.
- Expanded tests with new features and updated API documentation.
- Other small improvements and code clean up.
- Updated various backend packages.

Fixes #373

### CODE REVIEW

File: app/mikane/src/app/pages/account/change-password/change-password.component.ts [hunk 1]
```diff
@@ -36,8 +36,8 @@ export class ChangePasswordComponent implements OnInit, OnDestroy {
 
 	changePasswordForm = new FormGroup({
 		currentPassword: new FormControl<string>('', [Validators.required]),
-		newPassword: new FormControl<string>('', [Validators.required]),
-		newPasswordRetype: new FormControl<string>('', [Validators.required]),
+		newPassword: new FormControl<string>('', [Validators.required, Validators.minLength(3)]),
+		newPasswordRetype: new FormControl<string>('', [Validators.required, Validators.minLength(3)]),
```
Comment (aydex) [inline app/mikane/src/app/pages/account/change-password/change-password.component.ts:40]: Should probably have error messages in templte. Looks like there's only one for `invalid` now, but `required`, `minLength` and `match_error` exist

File: app/mikane/src/app/pages/register/register-user.component.ts [hunk 2]
```diff
@@ -54,8 +54,8 @@ export class RegisterUserComponent implements OnInit, AfterViewInit, OnDestroy {
 		email: new FormControl<string>('', [Validators.required, Validators.email], [emailValidator(this.formValidationService)]),
 		phone: new FormControl<string>('', [Validators.required], [phoneValidator(this.formValidationService)]),
 		passwordGroup: new FormGroup({
-			password: new FormControl<string>('', [Validators.required]),
-			passwordRetype: new FormControl<string>('', [Validators.required]),
+			password: new FormControl<string>('', [Validators.required, Validators.minLength(3)]),
```
Comment (aydex) [inline app/mikane/src/app/pages/register/register-user.component.ts:57]: Same thing here

File: app/mikane/src/app/pages/account/change-password/change-password.component.ts (hunk 1 above)
Comment (motionduck) [reply app/mikane/src/app/pages/account/change-password/change-password.component.ts:40]: Tried using `minLength`, but couldn't get it to work for some reason. Can try again.

Comment (aydex) [reply app/mikane/src/app/pages/account/change-password/change-password.component.ts:40]: Might be `minlength` (small L)

**Review by aydex — APPROVED** [review]

**Pull request merged**